### DISCUSSION TITLE
How to get context within the custom decorator.

### DISCUSSION THREAD

**Discussion by palashCItobuz** [MichalLytek/type-graphql#1067] [Q&A] [answered]
Implementing a custom method decorator which is supposed to be called after the `@Authorized()` decorator. However, I need the ctx with all values set in context in that decorator. How do we achieve something like this? Is there an example with return types etc? 
```
  @Authorized()
  @LogInstance()  // need ctx.user within this..
  @Query(() => UserCompanyRatings!)
  async hwsGetMeterValues(
```

```
import { createMethodDecorator } from 'type-graphql';
import { context } from '../context';

export function LogInstance<T extends object>() {
    return createMethodDecorator(async ({ args }, next) => {
        //context.prisma.meters.find({ ... })  some database query based on the user who has logged in.
        const instance = Object.assign({}, args);
        console.log(instance);
        return next();
    });
}

**Comment by MichalLytek** [comment, marked as answer]
Over here:
```ts
createMethodDecorator(async ({ args, context }, next)
```

**Reply by palashCItobuz** [reply]
@MichalLytek What should be the correct type instead of `<T extends object>`? Also I get this type validation warning when i'm trying to access user in it.
context.user
Property 'user' does not exist on type '{}'

**Reply by glitchgirl** [reply]
What did you do to get the user context?

**Reply by glitchgirl** [reply]
For anyone else who runs into this issue: 
```
async ({ context }: any
```
